contrib/dhalion/src/main/java/org/apache/dhalion/api/ISymptomDetector.java

```diff
@@ -0,0 +1,36 @@
+//  Copyright 2017 Twitter. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License
+package org.apache.dhalion.api;
+
+import java.util.Collection;
+
+import org.apache.dhalion.symptom.Symptom;
+
+public interface ISymptomDetector<T extends Symptom> extends AutoCloseable {
```

Comment:
The relation between a `symptom` and a `diagnoser` is not 1-1. Separating enables reuse and customization.